build-runtime.sh.in

```diff
@@ -1,19 +1,31 @@
 #! /bin/bash
 
+########################
+# build runtime script #
+########################
+#
+# template to be configured by the CMake build system
+
 if which clang >/dev/null; then
     CC=clang
 else
     CC=gcc
 fi
 
-set -x
+STRIP=strip
+
+set -xe
 
 build_dir=$(mktemp -d)
 orig_cwd=$(pwd)
-repo_root=$(dirname $(readlink -f $0))
+repo_root="@PROJECT_SOURCE_DIR@"
+
+# TODO: check whether those options are needed for other tools' build processes
+small_FLAGS="-Os -ffunction-sections -fdata-sections"
+small_LDFLAGS="-s -Wl,--gc-sections"
 
 cd $repo_root
-git_version=$(git describe --tags --always --abbrev=7)
+git_version="@GIT_VERSION@"
```

Comment:
Same here - why use a cryptic variable rather than a clear-text command? Where is it defined?

Reply:
The variable is defined by CMake, to avoid duplicate code, and ensure the _exact same value_ is used across the entire build system.